components/movementsensor/merged/merged.go

```diff
@@ -0,0 +1,373 @@
+			return err
+		}
+		m.angVelReporting = true
+	}
+
+	if laccChanged {
+		m.linAcc, err = firstGoodSensorWithProperties(deps,
+			newConf.LinearAcceleration,
+			&movementsensor.Properties{LinearAccelerationSupported: true})
+		if err != nil {
+			return err
+		}
+		m.linAccReporting = true
+	}
+
+	return nil
+}
+
+func (m *merged) Position(ctx context.Context, extra map[string]interface{}) (*geo.Point, float64, error) {
+	if m.pos == nil {
+		return geo.NewPoint(math.NaN(), math.NaN()), math.NaN(),
+			movementsensor.ErrMethodUnimplementedPosition
+	}
+
+	return m.Position(ctx, extra)
+}
+
+func (m *merged) Orientation(ctx context.Context, extra map[string]interface{}) (spatialmath.Orientation, error) {
+	if m.ori == nil {
+		nanOri := spatialmath.NewOrientationVector()
+		nanOri.OX = math.NaN()
+		nanOri.OY = math.NaN()
+		nanOri.OZ = math.NaN()
+		nanOri.Theta = math.NaN()
+		return nanOri,
+			movementsensor.ErrMethodUnimplementedOrientation
+	}
+	return m.Orientation(ctx, extra)
+}
+
+func (m *merged) CompassHeading(ctx context.Context, extra map[string]interface{}) (float64, error) {
+	if m.compass == nil {
+		return math.NaN(),
+			movementsensor.ErrMethodUnimplementedCompassHeading
+	}
+	return m.compass.CompassHeading(ctx, extra)
+}
+
+func (m *merged) LinearVelocity(ctx context.Context, extra map[string]interface{}) (r3.Vector, error) {
+	if m.linVel == nil {
+		return r3.Vector{X: math.NaN(), Y: math.NaN(), Z: math.NaN()},
+			movementsensor.ErrMethodUnimplementedLinearVelocity
+	}
+	return m.linVel.LinearVelocity(ctx, extra)
+}
+
+func (m *merged) AngularVelocity(ctx context.Context, extra map[string]interface{}) (spatialmath.AngularVelocity, error) {
+	if m.angVel == nil {
+		return spatialmath.AngularVelocity{X: math.NaN(), Y: math.NaN(), Z: math.NaN()},
+			movementsensor.ErrMethodUnimplementedAngularVelocity
+	}
+	return m.angVel.AngularVelocity(ctx, extra)
+}
+
+func (m *merged) LinearAcceleration(ctx context.Context, extra map[string]interface{}) (r3.Vector, error) {
+	if m.linAcc == nil {
+		return r3.Vector{X: math.NaN(), Y: math.NaN(), Z: math.NaN()},
+			movementsensor.ErrMethodUnimplementedLinearAcceleration
+	}
+	return m.linAcc.LinearAcceleration(ctx, extra)
+}
+
+func (m *merged) Accuracy(ctx context.Context, extra map[string]interface{}) (map[string]float32, error) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	accMap := make(map[string]float32)
+
+	if m.ori != nil {
+		oriAcc, err := m.ori.Accuracy(ctx, extra)
+		if err != nil {
+			oriAcc = map[string]float32{"err_orienation_accuracy": float32(math.NaN())}
+		}
+		maps.Copy(oriAcc, accMap)
+	}
+
+	if m.pos != nil {
+		posAcc, err := m.pos.Accuracy(ctx, extra)
+		if err != nil {
+			posAcc = map[string]float32{"err_position_accuracy": float32(math.NaN())}
+		}
+		maps.Copy(posAcc, accMap)
+	}
+
+	if m.compass != nil {
+		compassAcc, err := m.compass.Accuracy(ctx, extra)
+		if err != nil {
+			compassAcc = map[string]float32{"err_compass_accuracy": float32(math.NaN())}
```

Comment:
We'll see which sensor has the error, but not what the error was. I think we should log it in here. Similar comment in the other paragraphs, too.